plugin/plugin.go

```diff
@@ -29,44 +29,22 @@ import (
 	"github.com/notaryproject/notation-go/internal/slices"
 	"github.com/notaryproject/notation-go/log"
 	"github.com/notaryproject/notation-go/plugin/proto"
+	"github.com/notaryproject/notation-plugin-framework-go/plugin"
 )
 
 var executor commander = &execCommander{} // for unit test
 
 // GenericPlugin is the base requirement to be an plugin.
-type GenericPlugin interface {
```

Comment:
To mark an API deprecated, you need to update the documentation by adding `deprecated: <reason_and_recommendation>`.

See https://go.dev/wiki/Deprecated for more details.